app/controllers/admin/api/services/proxies_controller.rb

```diff
@@ -58,7 +58,11 @@ def show
   #
   def update
     if proxy.update_attributes(proxy_params)
-      proxy.deploy_v2 if proxy.apicast_configuration_driven
+      if proxy.service_mesh_integration?
```

Comment:
This decision seems to belong to `Proxy`.

Reply:
or to a "proxy deployment service".

Reply:
As commented in a previous comment I could use `deploy!` to handle that but that would be some changes that I am not going to do now. I think another PR is OK for refactoring stuff